sc-memory/cpp/utils/sc_log.hpp

```diff
@@ -29,24 +32,37 @@ class ScLog final
     Warning,
     Error,
     Python,
-    PythonError
+    PythonError,
+    Off
+  };
+
+  enum class OutputType: uint8_t
+  {
+    Console=0,
+    File
   };
 
-  _SC_EXTERN bool Initialize(std::string const & file_name, Type mode = Type::Info);
+  _SC_EXTERN bool Initialize(std::string const & file_name);
+
   _SC_EXTERN void Shutdown();
 
   /// TODO: thread safe
   _SC_EXTERN void Message(Type type, std::string const & msg, ScConsole::Color color = ScConsole::Color::White);
+
   _SC_EXTERN void SetMuted(bool value);
 
   _SC_EXTERN static ScLog * GetInstance();
 
 private:
   std::ofstream m_fileStream;
   Type m_mode;
+  OutputType m_output_mode;
   bool m_isMuted;
 
   static ScLog * ms_instance;
+
+  std::string const default_log_file = "system";
+  std::string const extension_log = ".log";
```

Comment:
Move extension to default_log_file constant. Make default_log_file constant as it usual we do in cpp(caps name, static and etc.)